receiver/signalfxreceiver/signalfxv2_to_metricdata.go

```diff
@@ -0,0 +1,237 @@
+	errSFxUnexpectedStringDatumType  = errors.New("datum value type of string is unexpected")
+	errSFxStringDatumNotNumber       = errors.New("datum string cannot be parsed to a number")
+	errSFxNoDatumValue               = errors.New("no datum value present for data-point")
+)
+
+// SignalFxV2ToMetricsData converts SignalFx proto data points to
+// consumerdata.MetricsData. Returning the converted data and the number of
+// dropped time series.
+func SignalFxV2ToMetricsData(
+	logger *zap.Logger,
+	sfxDataPoints []*sfxpb.DataPoint,
+) (*consumerdata.MetricsData, int) {
+
+	// TODO: not optimized at all, basically regenerating everything for each
+	// 	data point.
+	numDroppedTimeSeries := 0
+	md := &consumerdata.MetricsData{}
+	metrics := make([]*metricspb.Metric, 0, len(sfxDataPoints))
+	for _, sfxDataPoint := range sfxDataPoints {
+		if sfxDataPoint == nil {
+			// TODO: Log or metric for this odd ball?
+			continue
+		}
+
+		// First check if the type is convertible and the data point is consistent.
+		metricType, err := convertType(sfxDataPoint)
+		if err != nil {
+			numDroppedTimeSeries++
+			logger.Debug("SignalFx data-point type conversion error",
+				zap.Error(err),
+				zap.String("metric", sfxDataPoint.GetMetric()))
+			continue
+		}
+		point, err := buildPoint(sfxDataPoint, metricType)
+		if err != nil {
+			numDroppedTimeSeries++
+			logger.Debug("SignalFx data-point datum conversion error",
+				zap.Error(err),
+				zap.String("metric", sfxDataPoint.GetMetric()))
+			continue
+		}
+
+		labelKeys, labelValues := buildLabelKeysAndValues(sfxDataPoint.Dimensions)
+		descriptor := buildDescriptor(sfxDataPoint, labelKeys, metricType)
+		ts := &metricspb.TimeSeries{
+			// TODO: StartTimestamp can be set if each cumulative time series are
+			//  	tracked but right now it is not clear if it brings benefits.
+			//		Perhaps as an option so cost is "pay for play".
+			LabelValues: labelValues,
+			Points:      []*metricspb.Point{point},
+		}
+		metric := &metricspb.Metric{
+			MetricDescriptor: descriptor,
+			Timeseries:       []*metricspb.TimeSeries{ts},
+		}
+		metrics = append(metrics, metric)
+	}
+
+	md.Metrics = metrics
+	return md, numDroppedTimeSeries
+}
+
+func convertType(
+	sfxDataPoint *sfxpb.DataPoint,
+) (descType metricspb.MetricDescriptor_Type, err error) {
+
+	// Combine metric type with the actual data point type
+	sfxMetricType := sfxDataPoint.GetMetricType()
+	sfxDatum := sfxDataPoint.Value
+	if sfxDatum == nil {
+		return metricspb.MetricDescriptor_UNSPECIFIED, errSFxNilDatum
+	}
+
+	switch sfxMetricType {
+	case sfxpb.MetricType_GAUGE:
+		// Numerical: Periodic, instantaneous measurement of some state.
+		descType = metricspb.MetricDescriptor_GAUGE_DOUBLE
+		if sfxDatum.IntValue != nil {
+			descType = metricspb.MetricDescriptor_GAUGE_INT64
+		}
+
+	case sfxpb.MetricType_COUNTER:
+		// Numerical: Count of occurrences. Generally non-negative integers.
+		fallthrough
```

Comment:
Are we counting this cumulative counter as well? Would `case COUNTER, COMULATIVE_COUNTER` be better/more-obvious than fallthrough?